src/Common/tests/System/AssertExtensions.cs

```diff
@@ -24,8 +24,18 @@ public static void Throws<T>(string netCoreParamName, string netFxParamName, Act
             string expectedParamName =
                 RuntimeInformation.FrameworkDescription.StartsWith(".NET Framework") ?
                 netFxParamName : netCoreParamName;
-            
-            Assert.Equal(expectedParamName, exception.ParamName);
+
+            if (!RuntimeInformation.FrameworkDescription.StartsWith(".NET Native"))
```

Comment:
I'm fine add an IsNetNative property in PlatformDetection.cs but I wouldn't use it here, as it will require anyone including this extension file to also include the platformdections.cs file in the test project, which isn't ideal in my opinion.